File: trees/find_bottom_left_tree_value.py
from collections import deque

# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def findBottomLeftValue(self, root: Optional[TreeNode]) -> int:
        node_queue = deque([root])

        bottom_left_value = 0

        while node_queue:
            bottom_left_value = node_queue[0].val

            for _ in range(len(node_queue)):
                node = node_queue.popleft()

                if node.left:
                    node_queue.append(node.left)

                if node.right:
                    node_queue.append(node.right)

        return bottom_left_value

File: trees/find_bottom_left_tree_value.rb
def find_bottom_left_value(root)
    node_queue = [root]
    bottom_left_value = 0

    while !node_queue.empty?
        bottom_left_value = node_queue[0].val

        level_size = node_queue.size
        level_size.times do
            node = node_queue.shift

            node_queue << node.left if node.left
            node_queue << node.right if node.right
        end
      end
      
      bottom_left_value
end